### PR TITLE
New Serverless Pattern - API Gateway HTTP API to Step Functions Express Workflow [Java]

### PR DESCRIPTION
**Pull request by mridehalgh**
*Issue #, if available:*

*Description of changes:*
 Add a CDK implementation of a API Gateway HTTP API to a Step Functions Express Workflow.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by pputhran — CHANGES_REQUESTED** [review]
This pattern is missing example-pattern.json. Please refer to the sample template provided [here](https://github.com/aws-samples/serverless-patterns/tree/main/_pattern-model)

File: apigw-http-api-step-functions-express-java/example-pattern.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+  "title": "API Gateway to Step Functions",
```
Comment (pputhran) [inline apigw-http-api-step-functions-express-java/example-pattern.json:2]: add "API Gateway HTTP API to Step Functions"

File: apigw-http-api-step-functions-express-java/example-pattern.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "headline": "How it works",
+    "text": [
+      "This pattern explains how to deploy a CDK application with Amazon API Gateway that invokes a syncronous Step Functions workflow."
+    ]
+  },
+  "gitHub": {
+    "template": {
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/apigw-http-api-step-functions-express-java",
+      "projectFolder": "apigw-http-api-step-functions-express-java",
+      "templateFile": "apigw-http-api-step-functions-express-java/src/main/java/com/myorg/HttpApiGatewayStepFunctionsStack.java",
```
[the remark this answers is not in the document]
Comment (mridehalgh) [reply apigw-http-api-step-functions-express-java/example-pattern.json]: Fixed.

File: apigw-http-api-step-functions-express-java/example-pattern.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
+{
+  "title": "API Gateway to Step Functions",
+  "description": "Create an API Gateway HTTP APIs endpoint that starts a Step Functions synchronous Express Workflow.",
+  "language": "Java",
+  "level": "300",
+  "framework": "CDK",
+  "introBox": {
+    "headline": "How it works",
+    "text": [
+      "This pattern explains how to deploy a CDK application with Amazon API Gateway that invokes a syncronous Step Functions workflow."
```
[the remark this answers is not in the document]
Comment (mridehalgh) [reply apigw-http-api-step-functions-express-java/example-pattern.json]: Good spot

File: apigw-http-api-step-functions-express-java/example-pattern.json (hunk 1 above)
Comment (mridehalgh) [reply apigw-http-api-step-functions-express-java/example-pattern.json]: Done

**Review by pputhran — APPROVED** [review]

**Pull request merged**